File: Java/Tests/jdi-uitest-webtests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/RFileInputTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.core.interfaces.common.IFileInput;
import com.epam.jdi.uitests.core.interfaces.common.ILabel;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.web.robot.RFileInput;
import com.epam.web.matcher.base.BaseMatcher;
import com.epam.web.matcher.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;

import static com.codeborne.selenide.Condition.cssClass;
import static com.codeborne.selenide.Condition.visible;
import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.DATES_PAGE_FILLED;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.dates;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.*;
import static java.lang.String.format;

/**
 * Created by Dmitry_Lebedev1 on 11/12/2015.
 */
public class RFileInputTests extends InitTests {
    private IFileInput imageInput() { return dates.imageInput; }
    private RFileInput rImageInput() { return dates.rImageInput; }
    private ILabel uploadedFileName() { return dates.uploadedFileName; }

    @BeforeMethod
    public void before(final Method method) {
        isInState(DATES_PAGE_FILLED, method);
        BaseMatcher.setDefaultTimeout(10);
    }

    @Test
    public void inputTest() {
        rImageInput().input(getFPath());
        checkFileLoaded(fileName());
    }

    @Test
    public void sendKeysTest() {
        imageInput().sendKeys(getFPath());
        checkFileLoaded(fileName());
    }

    @Test
    public void newInputTest() {
        rImageInput().newInput(getFPath());
        checkFileLoaded(fileName());
    }


    private void checkFileLoaded(String fileName) {
        checkAction(format("FileUpload: file \"%s\" has been uploaded", fileName));
        Assert.areEquals(uploadedFileName().getText(), fileName);
    }

    @Test
    public void shouldTest(){
        rImageInput().shouldHave(cssClass("image-upload"))
                .shouldBe(visible);
    }
}

File: Python/Test/jdi_uitests_webtests/test/common/r_file_input_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.main.utils.common_action_data import CommonActionsData
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class RFileInputTests(InitTests):
    text_file = EpamJDISite.dates_page.r_image_input
    uploaded_file_name = EpamJDISite.dates_page.uploaded_file_name

    def setUp(self):
        Preconditions.DATES_PAGE.is_in_state()

    def test_input(self):
        self.text_file.input(CommonActionsData.get_file_path())
        self.check_file_loaded(CommonActionsData.get_file_name())

    def test_send_keys(self):
        self.text_file.send_keys(CommonActionsData.get_file_path())
        self.check_file_loaded(CommonActionsData.get_file_name())

    def test_new_input_test(self):
        self.text_file.new_input(CommonActionsData.get_file_path())
        self.check_file_loaded(CommonActionsData.get_file_name())

    def check_file_loaded(self, file_name):
        CommonActionsData.check_action('FileUpload: file "{0}" has been uploaded'.format(file_name))
        Assert.assert_contains(self.uploaded_file_name.get_text(), file_name)
